packages/fx-core/src/plugins/solution/fx-solution/debug/provisionLocal.ts

```diff
@@ -143,6 +143,24 @@ export async function setupLocalDebugSettings(
           }
         }
       }
+    } else if (inputs.platform === Platform.VS) {
+      if (includeFrontend) {
+        localSettings.frontend ??= {};
+        localSettings.frontend.tabEndpoint = "https://localhost:44302";
+        localSettings.frontend.tabDomain = "localhost";
+      }
+
+      if (includeBot) {
+        localSettings.bot ??= {};
+        const ngrokHttpUrl = await getNgrokHttpUrl(2544);
+        if (!ngrokHttpUrl) {
+          const error = NgrokTunnelNotConnected();
+          TelemetryUtils.sendErrorEvent(TelemetryEventName.setupLocalDebugSettings, error);
+          return err(error);
+        }
+        localSettings.bot.botEndpoint = ngrokHttpUrl;
+        localSettings.bot.botDomain = ngrokHttpUrl.slice(8);
```

Comment:
Nit: Kind of not robust - better to leverage some library to extract domain from url.

Reply:
Maybe we can add comment here to explain why use 8 as magic number